mopro-core/examples/halo2/keccak256/src/mopro.rs

```diff
@@ -0,0 +1,113 @@
+use crate::circuit::{pack_input_to_instance, unpack_input, KeccakCircuit};
+use crate::vanilla::KeccakConfigParams;
+use crate::{CIRCUIT_PARAMS, K, ROWS_PER_ROUND};
```

Comment:
I see what you mean of the logic to generate proof.

but can we not using `crate::circuit` but instead using like `halo2_keccak256_circuit::circuit...`
and you can maintain `halo2_keccak256_cirucit` in other repo
so we don't have to maintain the copy of keccak circuits

Reply:
Ah, I think I get it - the idea is to make the Keccak circuit leaner, right? We absolutely can do this, such as move the bulk of the logic somewhere else, and only use `halo2-circuit` to have an adapter from the `keccak` to work with `mopro` - effectively just keep the `mopro.rs` module as the rest can be placed outside. 

Do not think this will resolve the issue we have with the workspaces, but it will indeed simplify the code. Is this what you are suggesting this for?

Reply:
yes
so maybe we can easily control halo2 circuits by halo2 features
e.g. `--features halo2-fibonacci`, `--features halo2-keccak`